src/plugins/usage_collection/server/report/store_application_usage.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+import { Writable } from '@kbn/utility-types';
+import { ISavedObjectsRepository } from 'src/core/server';
+import { ApplicationUsageReport } from './schema';
+import { getDailyId } from '../../common/application_usage';
+
+type WritableApplicationUsageReport = Writable<ApplicationUsageReport>;
+
+export const storeApplicationUsage = async (
+  repository: ISavedObjectsRepository,
+  appUsages: ApplicationUsageReport[],
+  timestamp: Date
+) => {
+  if (!appUsages.length) {
+    return;
+  }
+
+  const dayId = getDayId(timestamp);
+  const aggregatedReports = aggregateAppUsages(appUsages);
+
+  return Promise.allSettled(
+    aggregatedReports.map(async (report) => incrementUsageCounters(repository, report, dayId))
+  );
+};
+
+const aggregateAppUsages = (appUsages: ApplicationUsageReport[]) => {
+  return [
+    ...appUsages
+      .reduce((map, appUsage) => {
+        const key = getKey(appUsage);
+        const aggregated: WritableApplicationUsageReport = map.get(key) ?? {
+          appId: appUsage.appId,
+          viewId: appUsage.viewId,
+          minutesOnScreen: 0,
+          numberOfClicks: 0,
+        };
```

Comment:
I wasn't sure we could receive multiple reports with the same `appId` / `viewId` tuple, but in doubt, I aggregate reports with the same 'key'